src/main/kotlin/io/github/enyason/client/ReplicateClient.kt

```diff
@@ -0,0 +1,40 @@
+package io.github.enyason.client
+
+import io.github.enyason.domain.models.Prediction
+import io.github.enyason.io.github.enyason.predictable.validate
+import io.github.enyason.predictable.Predictable
+import io.github.enyason.predictions.PredictionsApi
+import io.github.enyason.predictions.createPrediction
+
+class ReplicateClient(
+    private val predictionAPI: PredictionsApi
+) : Replicate {
+
+    override suspend fun createPrediction(predictable: Predictable): Result<Prediction> {
+        try {
+            predictable.validate()
+            val request = mapOf(
+                "version" to predictable.versionId,
+                "input" to predictable.input
+            )
+            val prediction = predictionAPI.createPrediction(request)
+            return Result.success(prediction)
+        } catch (error: Exception) {
+            return Result.failure(error)
+        }
+    }
+
+    override suspend fun getPrediction(predictionId: String): Result<Prediction> {
+        if (predictionId.isEmpty()) throw IllegalArgumentException("Provided an empty prediction ID")
```

Comment:
This can be extracted into a validate function and reused

Reply:
alright